accesshandler/pkg/providers/aws/flask/access.go

```diff
@@ -0,0 +1,582 @@
+
+	hasMore := true
+	var nextToken *string
+	log.Info("getting taskARN from task definition family", TaskDefinitionFamily)
+
+	//TODO: paginate all the list calls
+	for hasMore {
+		runningTasks, err := p.ecsClient.ListTasks(ctx, &ecs.ListTasksInput{Cluster: aws.String(p.ecsClusterARN.Get()), Family: &TaskDefinitionFamily, NextToken: nextToken})
+		if err != nil {
+			return "", err
+		}
+		describedTasks, err := p.ecsClient.DescribeTasks(ctx, &ecs.DescribeTasksInput{Cluster: aws.String(p.ecsClusterARN.Get()), Tasks: runningTasks.TaskArns})
+		if err != nil {
+			return "", err
+		}
+
+		//loop through all the tasks and find the latest version of the task definition
+		var revision int
+		var latestRevision string
+		for _, t := range describedTasks.Tasks {
+			if *t.LastStatus != "RUNNING" {
+				continue
+			}
+
+			tempVersion, err := strconv.Atoi(strings.Split(*t.TaskDefinitionArn, ":")[6])
+			if err != nil {
+				return "", err
+			}
+			if tempVersion > revision {
+				revision = tempVersion
+				latestRevision = *t.TaskArn
+			}
+			//return *t.TaskArn, nil
+			//exit the pagination
+			nextToken = runningTasks.NextToken
+			hasMore = nextToken != nil
+		}
+
+		return latestRevision, nil
+
+	}
+
+	//if nothing is found then we want to return an error
+	//will inform the user in the instructions of the not found error
+	return "", errTaskNotFound
+
+}
+
+// createPermissionSetAndAssignment creates a permission set with a name = grantID
+func (p *Provider) createPermissionSetAndAssignment(ctx context.Context, subject string, permissionSetName string, taskdefFamily string) (res *ssoadmin.CreateAccountAssignmentOutput, err error) {
+	//create  policy allowing for execute commands for the ecs task
+
+	clusterName := strings.Split(p.ecsClusterARN.Get(), "/")[1]
+
+	taskWildcard := fmt.Sprintf("arn:aws:ecs:%s:%s:task/%s/*", p.ecsRegion, p.awsAccountID, clusterName)
+	//task family arn building
+
+	tankFamilyARN := fmt.Sprintf("arn:aws:ecs:%s:%s:task-definition/%s:*", p.ecsRegion, p.awsAccountID, taskdefFamily)
+	//policy created:
+	//Resources:
+	// - ecs cluster arn
+	// - ecs task wildcard
+	// - ecs task definition wildcard for family
+	ecsPolicyDocument := policy.Policy{
+		Version: "2012-10-17",
+		Statements: []policy.Statement{
+			{
+				Effect: "Allow",
+				Action: []string{
+					"ecs:ExecuteCommand",
+					"ecs:DescribeTasks",
+				},
+
+				Resource: []string{taskWildcard, p.ecsClusterARN.Get(), tankFamilyARN},
+			},
+		},
+	}
+
+	// find the user ID from the provided email address.
+	user, err := p.getUser(ctx, subject)
+	if err != nil {
+		return nil, err
+	}
+	// create permission set with policy
+	permSet, err := p.ssoClient.CreatePermissionSet(ctx, &ssoadmin.CreatePermissionSetInput{
+		InstanceArn: aws.String(p.instanceARN.Get()),
+		Name:        aws.String(permissionSetName),
+		Description: aws.String("This permission set was automatically generated by Granted Approvals"),
```

Comment:
We can make this more descriptive

```suggestion
		Description: aws.String("Granted Approvals ECS Flask access"),
```